js/components/utils.js

```diff
@@ -432,37 +432,78 @@ var Utils = {
    * @function arrayDeepMerge
    * @param {Array} target - An array that will receive new items if additional items are passed
    * @param {Array} source - An array containing additional items to merge into target
-   * @param {Object} optionsArgument - optional parameters passed, i.e. arrayMerge function
+   * @param {Object} optionsArgument - optional parameters passed, i.e. arrayMerge, swap, unionBy, clone, arrayFusion
    * @returns {Array} new merged array with items from both target and source
    */
   arrayDeepMerge: function(target, source, optionsArgument) {
+    // replaces target with source, no merge
+    if (optionsArgument.arrayFusion === 'replace') {
```

Comment:
since optionsArgument is optional, this will throw an erro since it could be null.